xtask/src/tools/git.rs

```diff
@@ -106,20 +93,17 @@ impl GitRunner {
 
     // takes a PackageTag and kicks off a release in CircleCI
     pub(crate) fn tag_release(&self, package_tag: &PackageTag, dry_run: bool) -> Result<()> {
-        self.exec(&["pull"])?;
         if !dry_run {
-            // first, delete ALL local tags,
-            // it is possible a developer has tags
-            // locally that we do not want pushed to the remote
-            for local_tag in self.get_tags()? {
-                self.exec(&["tag", "-d", &local_tag])?;
-            }
             // create all the git tags we need from the PackageTag
             for tag in package_tag.all_tags() {
                 self.exec(&["tag", "-a", &tag, "-m", &tag]).context("If you want to re-publish this version, first delete the tag in GitHub at https://github.com/apollographql/federation-rs/current_git_tags")?;
             }
             // push up _only_ the tags that we just created
-            self.exec(&["push", "--tags", "--no-verify"])?;
+            for tag in package_tag.all_tags() {
```

Comment:
Is there a reason to create them all before pushing any? Or should we merge this loop with the one above it?

Reply:
99b17f4aca0c35d381d2b3bc9af51928c899cfa1